catalog/app/containers/Bucket/UploadDialog.js

```diff
@@ -937,3 +1074,34 @@ export default function UploadDialog({ bucket, open, onClose, refresh }) {
     </RF.Form>
   )
 }
+
+export default function UploadDialogWrapper({ bucket, open, onClose, refresh }) {
+  const s3 = AWS.S3.use()
+  const data = useData(requests.workflowsList, { s3, bucket })
+
+  return data.case({
+    Ok: (workflowsConfig) => (
+      <UploadDialog
+        {...{
+          bucket,
+          open,
+          onClose,
+          refresh,
+          workflowsConfig,
+        }}
+      />
+    ),
+    Err: () => (
```

Comment:
i think you should handle and log the error in the request function and just return null in case of error, that will allow you to just handle the `Ok` case and avoid repeating `<UploadDialog {...{ ... }} />` stuff